src/fun/counting_v2.go.tmpl

```diff
@@ -0,0 +1,271 @@
+{{/* end of configurable values */}}
+
+{{$db:=or 
+	(dbGet 0 "counting").Value 
+	(sdict "last" (sdict "user" .BotUser.ID "msg" 0) "next" 1 "highscore" (sdict "user" .BotUser.ID "num" 0 "time" currentTime) "saves" $saves)
+}}
+
+{{with .ExecData}}
+	{{$msg:=getMessage nil .}}
+	{{if not $msg}} {{/* check if message was deleted */}}
+		{{sendMessage nil (cembed 
+			"description" (printf "<@%d> deleted their expression which was correct!\nThe next number is %d" 
+				$db.last.user $db.next
+			) 
+			"color" 30654
+		)}}
+	{{else if $msg.EditedTimestamp}} {{/* check if message was edited */}}
+		{{sendMessage nil (cembed 
+			"description" (printf "<@%d> edited their expression, be careful" $db.last.user) 
+			"color" 30654
+		)}}
+	{{end}}
+	{{return}}
+{{end}}
+
+{{if ($re2:=reFindAllSubmatches (print `(?i)\A(?:-|<@!?` .BotUser.ID `>)\s*C(?:ount)?(?:ing)?Stat(?:istic)?s?\s*(?:(?:<@!?)?(\d{17,})|(l(?:eader)?b))?`) .Cmd)}}
+	{{if and (index $re2 0 1|eq "") (index $re2 0 2|eq "")}} {{/* general stats */}}
+		{{sendMessage nil (cembed 
+			"author" (sdict 
+				"icon_url" (.Guild.IconURL "512") 
+				"name" "🔢 Counting Statistics"
+			) 
+			"description" (printf "⌚ __Current Score__: %d\n🏅 __High Score__: %d on %v by `%s` (%d)\n⏮️ __Last Counter__: `%s` (%d)\n💾 __Saves Remaining__: %d" 
+				(sub $db.next 1) 
+				$db.highscore.num (formatTime $db.highscore.time "01/02") ($user:=userArg $db.highscore.user).Username $user.ID 
+				($user =userArg $db.last.user).Username $user.ID 
+				$db.saves
+			) 
+			"footer" (sdict "text" "Use this command: -CStats [User/\"Leaderboard\"]") 
+			"color" 30654
+		)}}
+		{{return}}
+	{{end}}
+
+	{{if ($user:=index $re2 0 1|toInt|userArg)}} {{/* user stats */}}
+		{{$userCount:=dbGet $user.ID "counting"}}
+		{{if not $userCount}}
+			{{sendMessage nil (cembed 
+				"title" (print "No available stats") 
+				"description" (printf "`%s` has yet to count ☹️\nMaybe give them a heads-up to come join?" 
+					$user.Username
+				) 
+				"footer" (sdict "text" "Use this command: -CStats [User: @/ID]") 
+				"color" 16711680
+			)}}
+		{{else}}
+			{{$userCount:=toInt $userCount.Value}}
+			{{$userCorrect:=(dbGet $user.ID "countingCorrect").Value}}
+			{{sendMessage nil (cembed 
+				"title" (printf "🔢 %s's Counting Statistics" $user.Username) 
+				"description" (printf "%s has counted **%d times**\n**%d** have been correct\nThis makes their average **%.2f%%**" 
+					$user.Mention $userCount (toInt $userCorrect) (div $userCorrect $userCount|mult 100.0)
+				) 
+				"footer" (sdict "text" "Use this command: -CStats [User: @/ID]") 
+				"color" 30654
+			)}}
+		{{end}}
+
+	{{else if eq (index $re2 0 2|lower) "leaderb" "lb"}} {{/* leaderboard */}}
+		{{$desc:=""}}{{$pos:=1}}
+		{{range dbTopEntries "countingCorrect" $leaderboardLength 0}}
+			{{- $desc =printf "%s\n#%-3d %4d - %-4s" $desc $pos (toInt .Value) (or (userArg .UserID) (str .UserID))}}
+			{{- $pos =add $pos 1 -}}
+		{{end}}
```

Comment:
```suggestion
		{{$desc:=""}}
		{{range $i,$v:=dbTopEntries "countingCorrect" $leaderboardLength 0}}
			{{- $desc =printf "%s\n#%-3d %4d - %-4s" $desc (add $i 1) (toInt $v.Value) (or $v.User.String (str $v.UserID))}}
		{{end}}
```
This uses the iteration index instead of manually incrementing `$pos` and saves a `userArg` call.

Reply:
I like the idea of using iteration index, however, the change to save the `userArg` call presents an issue. It seems that `or $v.User.String (str $v.UserID)` returns `#` for users not in the server. I'm not sure why this is; I didn't know entries made `.User.String` available as docs state "User object of type *discordgo.User* having only `.ID` field, .Mention is still usable with correct userID field entry."

Reply:
My bad, I misremembered that it isn't populated, `userArg` is needed then.